lib/vquic/quiche.c

```diff
@@ -237,10 +237,12 @@ static CURLcode quiche_has_connected(struct connectdata *conn,
   conn->httpversion = 30;
   conn->bundle->multiuse = BUNDLE_MULTIPLEX;
 
-  qs->h3config = quiche_h3_config_new(0, 1024, 0, 0);
+  qs->h3config = quiche_h3_config_new();
   if(!qs->h3config)
     return CURLE_OUT_OF_MEMORY;
 
+  quiche_h3_config_set_max_header_list_size(qs->h3config, 1024);
```

Comment:
I think we can omit this. Hardcoding the limit here is not very useful and AFAICT the ngtcp2 backend doesn't set a limit, so we should probably be consistent.

Reply:
Make sense. For anyone not tracking H3 specs closely, if an endpoint doesn't set a HTTP_MAX_HEADER_LIST_SIZE it is treated as effectively unlimited. Constructing the Config object and not setting anything will invoke that behaviour.